src/pylhe/__init__.py

```diff
@@ -369,6 +369,42 @@ def fromstring(cls, string):
         return cls(**dict(zip(cls.fieldnames, map(float, string.split()))))
 
 
+class LHEFile:
+    def __init__(self, init: LHEInit = None, events: Iterable[LHEEvent] = None):
+        self.init = init
+        self.events = events
+
+    def write(self, output_stream, rwgt=True, weights=False):
+        """
+        Write the LHE file to an output stream.
+        """
+        output_stream.write(
+            f"<LesHouchesEvents version=\"{self.init['LHEVersion']}\">\n"
+        )
+        output_stream.write(self.init.tolhe() + "\n")
+        for e in self.events:
+            output_stream.write(e.tolhe(rwgt=rwgt, weights=weights) + "\n")
+        output_stream.write("</LesHouchesEvents>")
+        return output_stream
+
+    def tolhe(self, rwgt=True, weights=False) -> str:
```

Comment:
I'm a bit on the fence, here. Given the way we write methods I would be tempted to have this as `to_lhe`.

Reply:
I see. OK, let's leave it, then.